lang/rust/avro/src/schema.rs

```diff
@@ -684,10 +692,11 @@ impl UnionSchema {
             Some((i, &self.schemas[i]))
         } else {
             // slow path (required for matching logical or named types)
-            self.schemas
-                .iter()
-                .enumerate()
-                .find(|(_, schema)| value.validate(schema))
+            self.schemas.iter().enumerate().find(|(_, schema)| {
```

Comment:
Why did you change this ?
This runs the same logic but now there won't be error log if the validation is not successful.

Reply:
I am going to revert this change since it is not related to the nullable records improvement and the build and tests still pass with the old code.
If you have a good reason to make this change we could still do it in a separate JIRA+PR.

Reply:
Since this cycles through all the variants to find a match, I was seeing a ton of error logs for every record even when there is no issue. Let me know is this makes sense, and I can reintroduce this change in a separate PR?

Reply:
I see!
Please create a new PR for the same JIRA that makes this change and has a unit test that shows the problem, i.e. asserts no error message is logged with the new behavior.
Thanks!